ListProcessingConsoleApp/ListProcessingConsoleApp/HomeworkFunctions.cs

```diff
@@ -0,0 +1,148 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ListProcessingConsoleApp
+{
+    public static class HomeworkFunctions
+    {
+        //Find the first element that matches a condition
+        public static int FindFirstEven(List<int> list)
+        {
+            return list
+                .Find(i => i % 2 == 0);
+        }
+
+        //Do all match a condition?
+        public static bool AreAllEven(List<int> list)
+        {
+            return list
+                .All(i => i % 2 == 0);
+        }
+
+        //Count how many match a condition
+        public static int NumberOfEven(List<int> list)
+        {
+            return list
+                .Where(i => i % 2 == 0)
+                .Count();
+        }
+
+        //Given a list of integers, make a list of integers where each integer is doubled.
+        //for loop
+        public static List<int> MultiplyListOfInts(List<int> list, int factor = 2)
+        {
+            List<int> outputList = new List<int>();
+            for (int i = 0; i < list.Count; i++)
+            {
+                outputList.Add(list[i] * factor);
+            }
+            return outputList;
+        }
+        //select/map
+        public static List<int> FunctionalMultiplyListOfInts(List<int> list, int factor = 2)
+        {
+            return list
+                .Select(i => i * factor)
+                .ToList();
+        }
+
+        //Given a list of Objects make a list where a property is set to the same value on each.
+        //for loop
+        public static List<Person> SetSameAge(List<Person> list, int age)
+        {
+            for (int i = 0; i < list.Count; i++)
+            {
+                list[i].Age = age;
+            }
+            return list;
+        }
+        //select/map
+        public static List<Person> FunctionalSetSameAge(List<Person> list, int age)
+        {
+            return (List<Person>)list.Select(p => p.Age = age);
```

Comment:
Does `.toList()` not work here?

Reply:
![image](https://user-images.githubusercontent.com/63496300/133076830-39cd67d4-82cd-4743-90cb-350d13efdfb0.png)

Reply:
It throws the above Exception, which I think is very similar to my attempt to cast explicitly